packages/block-editor/src/hooks/contrast-checker.js

```diff
@@ -20,14 +20,23 @@ function getBlockElementColors( blockEl ) {
 		return {};
 	}
 
+	// Special handling for button blocks - use the button link element for colors
+	let targetElement = blockEl;
+	if ( blockEl.classList.contains( 'wp-block-button' ) ) {
+		const buttonLink = blockEl.querySelector( '.wp-block-button__link' );
+		if ( buttonLink ) {
+			targetElement = buttonLink;
+		}
+	}
```

Comment:
The similar exceptional cases and hardcoded block classes don't scale. Why?

* What happens if the block classes or markup changes?
* What if other blocks also use a different target for applying colors? Do we try to hardcode all possible cases?

How I would handle a similar case: See how the block editor decides where to apply the colors. Check if there are already existing helpers we can reuse. Try to reuse existing methods and declarative settings for the contrast checker.

P.S. IIRC, special cases like this will use the `selectors` definition to target the correct elements.